### PR TITLE
Dart language syntax files

### PR DESCRIPTION
**Pull request by pr3d4t0r**
Dart language syntax files.  Licensed under Vim license.  Support for latest version of Dart, no support for Flutter by design; a future version may implement common framework names if there's demand.

Tested under the latest Vim fork as of 20191017.

Developer repository at https://github.com/pr3d4t0r/dart-vim-syntax/

Thanks Bram, Vim team!

### CODE REVIEW

**Comment by pr3d4t0r** [comment]
**Weird**.  I merged the syntax files with the current vim/vim:master upstream from my fork right before making this pull request, and did not touch the C files that fail coverage:

```
commit df3197b3a90e24baf096a7b83f634e73589f3502 (HEAD -> master, origin/master, origin/HEAD)
Merge: 9400d7168 a1cb1d1dc
Author: Eugene Ciurana <********@cime.net>
Date:   Thu Oct 17 19:33:46 2019 -0700

    Merge commit 'a1cb1d1dce14dd005797590721f1bcd0e7c3b35f'
    
    * commit 'a1cb1d1dce14dd005797590721f1bcd0e7c3b35f':
      patch 8.1.2171: mouse support not always available
      patch 8.1.2170: cannot build without the +termresponse feature

commit 9400d7168deffbdd2653c833b1ce549aa281d9f2 (working-copy)
Author: Eugene Ciurana <********@cime.net>
Date:   Thu Oct 17 19:31:25 2019 -0700

    Dart syntax files set up completed

commit a1cb1d1dce14dd005797590721f1bcd0e7c3b35f (tag: v8.1.2171, vim/master)
Author: Bram Moolenaar <Bram@vim.org>
Date:   Thu Oct 17 23:00:07 2019 +0200
```

Happy to pull from a different upstream master and make a new pull request if necessary.  Thanks and have a great weekend!

**Comment by dpelle** [comment]
@pr3d4t0r wrote:

> Weird. I merged the syntax files with the current vim/vim:master
> upstream from my fork right before making this pull request, and
> did not touch the C files that fail coverage

Don't worry about it. Coverage with Coveralls has noise somehow.
It's certainly not caused by your change. All tests passed which is
fine.  Thanks for the PR.

File: runtime/syntax/dart.vim [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+let s:cpo_save = &cpo
+set cpo&vim
+
+
+syn keyword dartCommentTodo     contained TODO FIXME XXX TBD
+syn match   dartLineComment     "//.*" contains=dartTodo,@Spell
+syn match   dartCommentSkip     "^[ \t]*\*\($\|[ \t]\+\)"
+syn region  dartComment         start="/\*"  end="\*/" contains=@Spell,dartTodo
+syn keyword dartReserved        assert async await class const export extends external final hide import implements interface library mixin on show super sync yield
+syn match   dartNumber          "-\=\<\d\+L\=\>\|0[xX][0-9a-fA-F]\+\>"
```
Comment (dpelle) [inline runtime/syntax/dart.vim:32]: You could replace [0-9a-fA-Z] with \x.
Matching can then be faster when using "set regexpengine=1" (old regex engine).

File: runtime/doc/syntax.txt [hunk 2]
```diff
@@ -1158,6 +1158,26 @@ startup vimrc: >
...
+used for developing mobile, desktop, web, and back-end applications.  Dart uses
+a C-like syntax derived from C, Java, and JavaScript, with features adopted
+from Smalltalk, Python, Ruby, and others.
+
+More information about the language and its development environment at the
+official Dart language website at https://dart.dev
+
+dart.vim syntax detects and highlights Dart statements, reserved words,
+type declarations, storage classes, conditionals, loops, interpolated values,
+and comments.  There is no support idioms from Flutter or any other Dart
```
Comment (dpelle) [inline runtime/doc/syntax.txt:1173]: Grammar is not right here.  Did you mean:  "There is no support **for** idioms from …"

**Review by dpelle — APPROVED** [review]

File: runtime/doc/syntax.txt (hunk 2 above)
Comment (pr3d4t0r) [reply runtime/doc/syntax.txt:1173]: Agh!  You're right -- the _for_ was lost in the edits.  Updating now.  Thanks for pointing this out.

**Comment by brammool** [comment]
Eugene Ciurana wrote:

> Dart language syntax files.  Licensed under Vim license.  Support for
> latest version of Dart, no support for Flutter by design; a future
> version may implement common framework names if there&#39;s demand.
> 
> Tested under the latest Vim fork as of 20191017.
> 
> Developer repository at https://github.com/pr3d4t0r/dart-vim-syntax/

Thanks.

+" Dart
+au BufRead,BufNewfile *.dart setf dart
+au BufRead,BufNewfile *.drt  set dart

This should probably be:

+au BufRead,BufNewfile *.dart,*.drt setf dart


+" Last Change:  See GitHub URL

That doesn't really work, since the file included with Vim won't be
automatically updated to the latest github version.

-- 
How come wrong numbers are never busy?

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Pull request closed without merging**